### ISSUE TITLE
Contacts directory and events directory: Filter by responsible surveillance officer does not work

### ISSUE THREAD

**Issue by StefanKock** [hzi-braunschweig/SORMAS-Project#3590]
### Bug Description

Filter by responsible surveillance officer does not work
Follow up to #3128: Contacts and events should be filterable by responsible surveillance officer

### Steps to Reproduce
Presumably like in #3128

### Expected Behavior

See #3128

### System Details
* SORMAS version: 1.51.0

### Additional Information
Was overlooked in #3128

**Comment by AndyBakcsy-she** [comment]
Verified issue on the latest version of SORMAS Version: 1.54.0-SNAPSHOT on the test environment

**Comment by AndyBakcsy-she** [comment]
While filtering on the Events page no results are displayed because of another issue: #3983

**Comment by MateStrysewske** [comment]
The list of selectable officers in the contact directory has two issues:
* It allows the user to select Surveillance Officers, but only Contact Officers should be selectable here
* The list of selectable officers doesn't directly update when setting a region and district, but only after "Apply filters" has been clicked; the list should update as soon as one of the jurisdiction filters is changed

**Comment by raghupola-vg** [comment]
verified on test environment with the latest deploy and the issue is Fixed.
